### PR TITLE
Reducing log level of restarting to WARN

### PR DESCRIPTION
**Pull request by zmarois**
Closes https://github.com/akka/akka/issues/25019

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

**Review by raboof — APPROVED** [review]
That seems reasonable, failing streams do not necessarily require operator attention.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala [hunk 1]
```diff
@@ -464,7 +464,7 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
         if (finishing || maxRestartsReached()) {
           fail(out, ex)
         } else {
-          log.error(ex, "Restarting graph due to failure")
+          log.warn(ex, "Restarting graph due to failure")
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: I'm afraid it's `log.warning` ;)

**Review by raboof — CHANGES_REQUESTED** [review]

**Comment by lightbend-cla-validator** [comment]
Hi @zmarois,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala (hunk 1 above)
Comment (zmarois) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: Darn, that's what I get for just letting your build system do the building.

**Comment by lightbend-cla-validator** [comment]
Hi @zmarois,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by zmarois** [comment]
As https://github.com/akka/akka/issues/16745#issuecomment-73368691 points out, exceptions cannot be provided to warnings (or info or debug). I'm left with just logging the exception message if we want to reduce the log level.

**Comment by raboof** [comment]
Ah, that is indeed unfortunate. As part of the work for the 'typed' logging API we did add this feature there and introduced some infrastructure (https://github.com/johanandren/akka/pull/5/files#diff-47ea75a55aa7f26d9b488bfb764f9d79R791), so this might be a good time to look into adding these variations again?

**Comment by raboof** [comment]
Added #25033 - if we can get that in we can keep the stack trace, which I do think seems useful in this case.

**Comment by zmarois** [comment]
I agree @raboof . Thanks a ton for going out of your way to do that for me! I'll update this if your change goes through (although you already did all the heavy lifting).

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala [hunk 2]
```diff
@@ -464,7 +464,7 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
         if (finishing || maxRestartsReached()) {
           fail(out, ex)
         } else {
-          log.error(ex, "Restarting graph due to failure")
+          log.warning("Restarting graph due to failure: {}", ex.getMessage)
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: good, I think the logging should be more adjustable for the restart operator, but that is another issue

**Comment by lightbend-cla-validator** [comment]
Hi @zmarois,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala [hunk 3]
```diff
@@ -464,7 +464,7 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
         if (finishing || maxRestartsReached()) {
           fail(out, ex)
         } else {
-          log.error(ex, "Restarting graph due to failure")
+          log.warning(ex, "Restarting graph due to failure")
```
Comment (zmarois) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: Build will fail until I pull in https://github.com/akka/akka/pull/25033.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala (hunk 2 above)
Comment (zmarois) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: I'd gladly make this more configurable in this change, but if you think its a separate concern, then I won't yet.

I can understand a general desire to make this more configurable, but I'm not too sure what specific variability clients want. My only desire is to vary the log level, and even that I don't really need variation, just reduction for all my current use cases.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala (hunk 3 above)
Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala]: Due to the hazzle of adding that in a binary compatible way we decided to not do that. Please change this to:
```
if (log.isWarningEnabled)
  log.warning("Restarting graph due to failure: {}", Logging.stackTraceFor(e))
```

**Comment by lightbend-cla-validator** [comment]
Hi @zmarois,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Review by patriknw — APPROVED** [review]
LGTM, after CLA signed

**Comment by zmarois** [comment]
CLA signed

**Review by raboof — APPROVED** [review]
Great contribution, thanks @zmarois!

**Comment by raboof** [comment]
(manually checked the CLA and it is :+1:)

**Pull request merged**